python/datasources/uhc.py

```diff
@@ -0,0 +1,122 @@
+import pandas as pd
+
+from ingestion.standardized_columns import Race
+import ingestion.standardized_columns as std_col
+
+from datasources.data_source import DataSource
+from ingestion import gcs_to_bq_util, constants
+
+UHC_RACE_GROUPS = [
+    'American Indian/Alaska Native',
+    'Asian',
+    'Black',
+    'Hawaiian/Pacific Islander',
+    'Hispanic',
+    'Multiracial',
+    'Other Race',
+    'White',
+    'All',
+]
+
+UHC_AGE_GROUPS = ['18-44', '45-64', '65+', 'All']
+
+UHC_SEX_GROUPS = ['Male', 'Female', 'All']
+
+UHC_RACE_GROUPS_TO_STANDARD = {
+    'American Indian/Alaska Native': Race.AIAN_NH.value,
+    'Asian': Race.ASIAN_NH.value,
+    'Black': Race.BLACK_NH.value,
+    'Hispanic': Race.HISP.value,
+    'Hawaiian/Pacific Islander': Race.NHPI_NH.value,
+    'Other Race': Race.OTHER_STANDARD_NH.value,
+    'White': Race.WHITE_NH.value,
+    'Multiracial': Race.MULTI_NH.value,
+    'All': Race.ALL.value,
+}
+
+BASE_UHC_URL = "https://www.americashealthrankings.org/api/v1/downloads/210"
+
+UHC_DETERMINANTS_OF_HEALTH = {
+    "Chronic Obstructive Pulmonary Disease": std_col.COPD_PCT,
+    "Diabetes": std_col.DIABETES_PCT,
+}
+
+
+class UHCData(DataSource):
+
+    @staticmethod
+    def get_id():
+        return 'UHC_DATA'
+
+    @staticmethod
+    def get_table_name():
+        return 'uhc_data'
+
+    def upload_to_gcs(self, _, **attrs):
+        raise NotImplementedError(
+            'upload_to_gcs should not be called for UHCData')
+
+    def write_to_bq(self, dataset, gcs_bucket, **attrs):
+        df = gcs_to_bq_util.load_csv_as_dataframe_from_web(
+                BASE_UHC_URL, dtype={std_col.STATE_FIPS_COL: str}
+            )
+
+        for b in [std_col.RACE_OR_HISPANIC_COL, std_col.AGE_COL, std_col.SEX_COL]:
+            breakdown_df = self.generate_breakdown(b, df)
+
+            column_types = {c: 'STRING' for c in breakdown_df.columns}
+            for col in [std_col.COPD_PCT, std_col.DIABETES_PCT]:
+                column_types[col] = 'FLOAT'
+
+            if std_col.RACE_INCLUDES_HISPANIC_COL in breakdown_df.columns:
+                column_types[std_col.RACE_INCLUDES_HISPANIC_COL] = 'BOOL'
+
+            breakdown_df.to_csv("test_%s.csv" % b)
+            gcs_to_bq_util.add_dataframe_to_bq(
+                breakdown_df, dataset, b, column_types=column_types)
+
+    def generate_breakdown(self, breakdown, df):
+        breakdown_map = {
```

Comment:
This seems like something that should be a top-level constant rather than defined in a method